src/server/metrics.rs

```diff
@@ -167,18 +175,21 @@ pub fn metrics_from_req(req: &HttpRequest) -> Result<Box<StatsdClient>, Error> {
         .clone())
 }
 
-/// Create a cadence StatsdClient from the given options
-pub fn metrics_from_opts(opts: &Settings) -> Result<StatsdClient, ApiError> {
-    let builder = if let Some(statsd_host) = opts.statsd_host.as_ref() {
+pub fn metrics_from_opts(
+    label: String,
+    host: Option<String>,
```

Comment:
not terribly important but this could take references instead. `Option::as_deref` can easily give you an `Option<&str>` from the host field `Option<String>`

```suggestion
    label: &str
    host: Option<&str>,
```

Reply:
Nice catch!